risc0/circuit/rv32im/benches/eval_check.rs

```diff
@@ -38,7 +38,7 @@ pub fn eval_check(c: &mut Criterion) {
     #[cfg(feature = "cuda")]
     for po2 in [2, 8, 16, 20, 21].iter() {
         let params = EvalCheckParams::new(*po2);
-        let hal = std::rc::Rc::new(risc0_zkp::hal::cuda::CudaHalSha256::new());
+        let hal = std::sync::Arc::new(risc0_zkp::hal::cuda::CudaHalSha256::new());
```

Comment:
Did we switch to a Arc because the hal now cross thread boundaries?

Reply:
Initially I tried to make a singleton `OnceCell` registry of all provers, but ran into trouble with getting this to work with rustacuda. Eventually I replaced rustacude with cust, which fixed those issues, but by this point I abandonded the singleton approach. Now there's no reason to use `Arc` so I'll revert back to `Rc`.